File: apps/xunit-example/StoryOptionsStoryTest.cs
using ExecutableStories.Xunit;
using Xunit;

namespace XunitExample.Tests;

public class StoryOptionsStoryTest
{
    [Fact]
    public void Story_with_single_tag()
    {
        Story.Init("Story with single tag", "smoke");
        Story.Note("Single tag for basic categorization");
        Story.Given("a tagged story");
        Story.When("tests are filtered");
        Story.Then("this story matches the 'smoke' tag");
        Story.RecordAndClear();
    }

    [Fact]
    public void Story_with_multiple_tags()
    {
        Story.Init("Story with multiple tags", "smoke", "regression", "critical");
        Story.Note("Multiple tags for flexible filtering");
        Story.Given("a story with multiple tags");
        Story.When("tests are filtered by any tag");
        Story.Then("this story matches multiple filters");
        Story.RecordAndClear();
    }

    [Fact]
    public void Story_with_feature_tags()
    {
        Story.Init("Story with feature tags", "feature:auth", "feature:login");
        Story.Note("Tags can use prefixes for organization");
        Story.Given("a story tagged by feature");
        Story.Then("feature filtering is possible");
        Story.RecordAndClear();
    }
}

File: apps/junit5-example/src/test/kotlin/example/StoryOptionsStoryTest.kt
package example

import dev.executablestories.junit5.Story
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertTrue

class StoryOptionsStoryTest {

    @Test
    fun storyWithSingleTag() {
        Story.init("Story with single tag", "smoke")
        Story.note("Single tag for basic categorization")
        Story.given("a tagged story")
        Story.`when`("tests are filtered")
        Story.then("this story matches the 'smoke' tag")
        assertTrue(true)
    }

    @Test
    fun storyWithMultipleTags() {
        Story.init("Story with multiple tags", "smoke", "regression", "critical")
        Story.note("Multiple tags for flexible filtering")
        Story.given("a story with multiple tags")
        Story.`when`("tests are filtered by any tag")
        Story.then("this story matches multiple filters")
        assertTrue(true)
    }

    @Test
    fun storyWithFeatureTags() {
        Story.init("Story with feature tags", "feature:auth", "feature:login")
        Story.note("Tags can use prefixes for organization")
        Story.given("a story tagged by feature")
        Story.then("feature filtering is possible")
        assertTrue(true)
    }

    @Test
    fun storyWithOnlyTags() {
        Story.init("Story with only tags option", "minimal")
        Story.given("story with only tags option")
        Story.then("other options are optional")
        assertTrue(true)
    }
}
